File: scripts/panas_rea_calculator.py
#!/usr/bin/env python3
import argparse

def compute_vbt(v_inmueble: float, v_com: float, v_local_usdt: float, v_plus: float,
                w1: float = 0.45, w2: float = 0.25, w3: float = 0.20, w4: float = 0.10) -> float:
    return w1 * v_inmueble + w2 * v_com + w3 * v_local_usdt + w4 * v_plus

def recommend_ltv(base: float, bonus_operativo: float, bonus_liquidez: float, penalidad_riesgo: float,
                  cap: float = 0.80, base_floor: float = 0.70) -> float:
    bruto = base_floor + bonus_operativo + bonus_liquidez - penalidad_riesgo
    return max(0.0, min(cap, bruto))

def main():
    p = argparse.ArgumentParser(description="PANAS-REA Calculator")
    p.add_argument("precio", type=float, help="Precio de referencia")
    p.add_argument("avaluo", type=float, help="Avalúo técnico (V_inmueble)")
    p.add_argument("comparables", type=float, help="Comparables €/m² (V_com)")
    p.add_argument("fiat_usdt", type=float, help="Paridad FIAT→USDT (V_local→USDT)")
    p.add_argument("plusvalia", type=float, help="Plusvalía 12–24m (V_plus)")
    p.add_argument("--bonus_operativo", type=float, default=0.05)
    p.add_argument("--bonus_liquidez", type=float, default=0.02)
    p.add_argument("--penalidad_riesgo", type=float, default=0.05)
    p.add_argument("--cap", type=float, default=0.80)
    p.add_argument("--base_floor", type=float, default=0.70)
    args = p.parse_args()

    vbt = compute_vbt(args.avaluo, args.comparables, args.fiat_usdt, args.plusvalia)
    ltv = recommend_ltv(args.base_floor, args.bonus_operativo, args.bonus_liquidez, args.penalidad_riesgo, cap=args.cap, base_floor=args.base_floor)
    emision = vbt * ltv

    print(f"VBT: {vbt:.2f}")
    print(f"LTV recomendado: {ltv:.2%}")
    print(f"Monto de emisión: {emision:.2f}")

if __name__ == "__main__":
    main()

File: scripts/panas-rea-calculator.mjs
#!/usr/bin/env node
import process from 'node:process';

function computeVBT(v_inmueble, v_com, v_local_usdt, v_plus, w1=0.45, w2=0.25, w3=0.20, w4=0.10) {
  return w1 * v_inmueble + w2 * v_com + w3 * v_local_usdt + w4 * v_plus;
}

function recommendLTV(base, bonusOperativo, bonusLiquidez, penalidadRiesgo, cap=0.80, baseFloor=0.70) {
  const bruto = baseFloor + bonusOperativo + bonusLiquidez - penalidadRiesgo;
  return Math.max(0, Math.min(cap, bruto));
}

function main() {
  const [precio, avaluo, comparables, fiat_usdt, plusvalia] = process.argv.slice(2).map(Number);
  if ([precio, avaluo, comparables, fiat_usdt, plusvalia].some((v) => !Number.isFinite(v))) {
    console.error("Uso: panas-rea-calculator.mjs <precio> <avaluo> <comparables> <fiat_usdt> <plusvalia>");
    process.exit(1);
  }
  const bonusOperativo = Number(process.env.PANAS_REA_OP_HISTORY_BONUS || 0.05);
  const bonusLiquidez = Number(process.env.PANAS_REA_POOL_LIQ_BONUS || 0.02);
  const penalidadRiesgo = Number(process.env.PANAS_REA_REGION_RISK_PENALTY || 0.05);
  const cap = Number(process.env.PANAS_REA_LTV_CAP || 0.80);
  const baseFloor = Number(process.env.PANAS_REA_LTV_BASE || 0.70);

  const vbt = computeVBT(avaluo, comparables, fiat_usdt, plusvalia);
  const ltv = recommendLTV(baseFloor, bonusOperativo, bonusLiquidez, penalidadRiesgo, cap, baseFloor);
  const emision = vbt * ltv;

  console.log(`VBT: ${vbt.toFixed(2)}`);
  console.log(`LTV recomendado: ${(ltv*100).toFixed(2)}%`);
  console.log(`Monto de emisión: ${emision.toFixed(2)}`);
}

main();


